File: erlang_bench/bench.erl
% Erlang benchmark
-module(bench).
-export([main/1, benchmark/1]).

% Calculate the aggregate of the sums of the sublists
benchmark(List) ->
    Sum = fun(A,B)-> A + B end,
    Accum = fun(L) -> lists:foldr(Sum, 0, L) end,
    Accum(lists:map(Accum, List)).

% Generate a list of random intenger lists
generate_list(ListSize, NumberSize) ->
    [[ random:uniform(100) || _  <- lists:seq(1, NumberSize)] || _ <- lists:seq(1,ListSize)].

% Execute main method
main(Args) ->
    % Parse input args
    [Arg1 , Arg2 | _] = Args,
    {ListSize, _} = string:to_integer(atom_to_list(Arg1)),
    {NumberSize, _} = string:to_integer(atom_to_list(Arg2)),

    % Call the algorithm
    List = generate_list(ListSize, NumberSize),
    {ElapsedMicro, Result} = timer:tc(bench, benchmark, [List]),
    Elapsed = ElapsedMicro / 1000,
    io:fwrite("[Erlang ! Array Sum]: Elapsed time: ~p msec ( Result: ~p )\n", [Elapsed, Result]).

File: elixir_bench/bench.ex
defmodule Benchmark do
    def generate_list(numberLists, numberItems) do
        lc _ inlist Enum.to_list(0 .. numberLists) do
            lc _ inlist Enum.to_list(0 .. numberItems) do
               :random.uniform(100)
            end
        end
    end

    defp calculate_list(list) do
        List.foldr(list, 0, &(&1 + &2))
    end

    def do_bench(input) do
        calculate_list Enum.map input, &calculate_list/1
    end

    def main(args) do
        [arg1, arg2 | _] = args
        {numberLists, _} = Integer.parse(arg1)
        {numberItems, _} = Integer.parse(arg2)
        list = generate_list(numberLists, numberItems)
        {elapsed, result} = :timer.tc(&do_bench/1, [list])
        IO.puts "[Elixir ! Array Sum] Elapsed time: #{elapsed/1000} msec ( Result: #{result} )"
    end
end

Benchmark.main(System.argv())
